### PR TITLE
Martin/ios 552 add attachment expanded preview

### PR DESCRIPTION
**Pull request by MartinJibber**
https://user-images.githubusercontent.com/88675954/159803483-1bf7f894-8d34-46d7-b4fa-56acba343a49.mp4

### CODE REVIEW

File: iOS/Flows/Conversation/Attachments/AttachmentsCoordinator.swift [hunk 1]
```diff
@@ -9,7 +9,7 @@
 import Foundation
 
 enum AttachmentOption {
-    case attachements([Attachment])
+    case attachments([Attachment])
```
Comment (MartinJibber) [inline iOS/Flows/Conversation/Attachments/AttachmentsCoordinator.swift:12]: Spelling

File: iOS/Flows/Conversation/Cells/ConversationMessagesCell.swift [hunk 2]
```diff
@@ -263,8 +263,8 @@ class ConversationMessagesCell: UICollectionViewCell, ConversationUIStateSettabl
         return self.collectionView.convert(dropZoneFrame, to: targetView)
     }
 
-    func getBottomFrontmostCell() -> MessageCell? {
-        return self.collectionLayout.getBottomFrontmostCell()
+    func getFrontmostCell() -> MessageCell? {
+        return self.collectionLayout.getFrontmostCell()
```
Comment (MartinJibber) [inline iOS/Flows/Conversation/Cells/ConversationMessagesCell.swift:267]: There's no longer two sections so the "bottom cell" doesn't make sense any more.

File: iOS/Flows/Conversation/Cells/MessagesTimeMachineCollectionViewLayout.swift [hunk 3]
```diff
@@ -87,13 +87,6 @@ class MessagesTimeMachineCollectionViewLayout: TimeMachineCollectionViewLayout {
         return CGPoint(x: 0, y: self.maxZPosition)
     }
 
-    private func getCenterOfItems() -> CGPoint {
-        let topCenter = self.getItemCenterPoint(in: 0, withYOffset: 0, scale: 1.0)
-        let bottomCenter = self.getItemCenterPoint(in: 1, withYOffset: 0, scale: 1.0)
-        let center = CGPoint(x: topCenter.x, y: (topCenter.y + bottomCenter.y) / 2)
-        return center
-    }
```
Comment (MartinJibber) [inline iOS/Flows/Conversation/Cells/MessagesTimeMachineCollectionViewLayout.swift:95]: Unused.

File: iOS/Flows/Conversation/Cells/TimeMachineCollectionViewLayout.swift [hunk 4]
```diff
@@ -253,12 +251,12 @@ class TimeMachineCollectionViewLayout: UICollectionViewLayout {
 
     // MARK: - Attribute Helpers
 
-    /// Gets the index path of the frontmost item in the given section.
-    func getFrontmostIndexPath(in section: SectionIndex) -> IndexPath? {
```
Comment (MartinJibber) [inline iOS/Flows/Conversation/Cells/TimeMachineCollectionViewLayout.swift:257]: No need to specify a section. There's only one.

File: iOS/Flows/Conversation/Conversation List/ConversationListCoordinator.swift [hunk 5]
```diff
@@ -32,19 +31,13 @@ class ConversationListCoordinator: InputHandlerCoordinator<Void> {
                                                 startingMessageID: startingMessageId)
 
         super.init(with: vc, router: router, deepLink: deepLink)
+
+        vc.messageCellDelegate = self
```
Comment (MartinJibber) [inline iOS/Flows/Conversation/Conversation List/ConversationListCoordinator.swift:35]: This delegate assignment replaces the handler closure below.

File: iOS/Flows/Conversation/Conversation List/ConversationListCoordinator.swift [hunk 6]
```diff
@@ -134,26 +127,36 @@ extension ConversationListCoordinator {
     }
 }
 
-// MARK: - Image View Flow
-
-extension ConversationListCoordinator: LightboxControllerDismissalDelegate {
+// MARK: - MessageCellDelegate
 
-    func presentImageFlow(for imageURL: URL) {
```
Comment (MartinJibber) [inline iOS/Flows/Conversation/Conversation List/ConversationListCoordinator.swift:141]: Image flow moved into the base class.

File: iOS/Flows/Conversation/Conversation List/ConversationListCoordinator.swift [hunk 7]
```diff
@@ -134,26 +127,36 @@ extension ConversationListCoordinator {
...
+        let message = Message.message(with: messageInfo.0, messageId: messageInfo.1)
+
+        switch message.kind {
+        case .photo(photo: let photo, _):
+            guard let url = photo.url else { return }
 
+            self.presentImageFlow(for: [url], startingURL: url)
+        case .text, .attributedText, .location, .emoji, .audio, .contact, .link, .video:
+            break
+        }
```
Comment (MartinJibber) [inline iOS/Flows/Conversation/Conversation List/ConversationListCoordinator.swift:160]: Currently only supporting photo attachments. Links are up next.

File: iOS/Flows/Conversation/Conversation List/ConversationListViewController+Extensions.swift [hunk 8]
```diff
@@ -12,8 +12,6 @@ import StreamChat
 extension ConversationListViewController {
 
     func setupInputHandlers() {
-        self.dataSource.messageCellDelegate = self
```
Comment (MartinJibber) [inline iOS/Flows/Conversation/Conversation List/ConversationListViewController+Extensions.swift:15]: Unneeded, the coordinator will handle message cell events.

File: iOS/Flows/Conversation/Conversation List/ConversationListViewController.swift [hunk 9]
```diff
@@ -212,12 +212,13 @@ class ConversationListViewController: InputHandlerViewContoller, ConversationLis
         // Determine if this is a reply message or regular message. If it's a reply, select the parent
         // message so we can open the thread experience.
         if let parentMessageId = message.parentMessageId {
-            await messagesCell.scrollToMessage(with:  parentMessageId, animateSelection: true)
-            self.selectedMessageView = messagesCell.getBottomFrontmostCell()?.content
```
Comment (MartinJibber) [inline iOS/Flows/Conversation/Conversation List/ConversationListViewController.swift:216]: I got rid of having to manually set the selected message view. Instead we just assume that the front most message is the one that's selected. See the transitionable code below.

File: iOS/Flows/Image Viewer/ImageViewCoordinator.swift [hunk 10]
```diff
@@ -0,0 +1,69 @@
...
+        self.imageURLs = imageURLs
+        self.startURL = startURL
+
+        super.init(router: router, deepLink: deepLink)
+    }
+}
+
+// MARK: - ImageViewController
+
+class ImageViewController: LightboxController, Dismissable {
```
Comment (MartinJibber) [inline iOS/Flows/Image Viewer/ImageViewCoordinator.swift:56]: Dismissable version of the lightbox controller.

File: iOS/Flows/Image Viewer/ImageViewCoordinator.swift [hunk 11]
```diff
@@ -0,0 +1,69 @@
...
+//  Jibber
+//
+//  Created by Martin Young on 3/23/22.
+//  Copyright © 2022 Benjamin Dodgson. All rights reserved.
+//
+
+import Foundation
+import Lightbox
+
+class ImageViewCoordinator: PresentableCoordinator<Void> {
```
Comment (MartinJibber) [inline iOS/Flows/Image Viewer/ImageViewCoordinator.swift:12]: Basic presentable coordinator for the image viewer flow.

**Review by Benjamindodgson — APPROVED** [review]
Looks dope. May conflict with what I'm working on. Ill let you know.

**Pull request merged**